src/main/java/seedu/bigpp/command/buildercommand/BuilderListComponentCommand.java

```diff
@@ -20,27 +35,127 @@ public BuilderListComponentCommand(String arguments) {
+                    }
+                    if (Float.parseFloat(priceFrom) > Float.parseFloat(priceTo)) {
+                        throw new BuilderPriceLogicException();
+                    }
+                    if (Float.parseFloat(priceFrom) <= 0 || Float.parseFloat(priceTo) <= 0) {
+                        throw new BuilderInvalidPriceNumber();
+                    }
+                    componentList = filterByPrice(componentList, priceFrom, priceTo);
+                    flagDescriptionArray.add("price: from " + priceFrom + " to " + priceTo);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new BuilderInvalidNumberFlags();
+                }
+                break;
+            default:
+                break;
+            }
         }
+        String outputString = "";
 
-        String outputString = "Here are all available components of type '" + componentTypeString + "': \n";
+        if (componentList.size() == 0) {
+            outputString = "There are no components of type '" + componentType + "' ";
+        } else {
+            outputString = "Here are all available components of type '" + componentType + "': \n";
+        }
+
+        if (flagDescriptionArray.size() > 0) {
+            outputString += "meeting the following criteria: \n";
+            for (String flagDescription : flagDescriptionArray) {
+                outputString += flagDescription + "\n";
+            }
+        }
 
-        return outputString + dataStorage.stringToComponentListMap.get(componentTypeString).getListString();
+        return outputString + componentList.getListString();
+    }
+
+    private ComponentList<?> filterByName(ComponentList<?> componentList, String name) {
+        ComponentList<?> filteredComponentList = new ComponentList<>();
+        for (Component component : componentList) {
+            if (component.getName().toLowerCase().contains(name)) {
+                filteredComponentList.add(component);
+            }
+        }
+        return filteredComponentList;
+    }
+
+    private ComponentList<?> filterByBrand(ComponentList<?> componentList, String brand) {
+        ComponentList<?> filteredComponentList = new ComponentList<>();
+        for (Component component : componentList) {
+            if (component.getBrand().toLowerCase().contains(brand)) {
+                filteredComponentList.add(component);
+            }
+        }
+        return filteredComponentList;
+    }
+
+    private ComponentList<?> filterByPrice(ComponentList<?> componentList, String priceFrom, String priceTo) {
+        ComponentList<?> filteredComponentList = new ComponentList<>();
+        for (Component component : componentList) {
+            if (component.getPrice() >= Float.parseFloat(priceFrom) && component.getPrice() <= Float.parseFloat(
+                    priceTo)) {
+                filteredComponentList.add(component);
+            }
+        }
+        return filteredComponentList;
```

Comment:
Could we possibly add these to ComponentList Class instead, abstracting the methods from this file?

Reply:
yes, done!